Server/Classes/Handlers/VehicleSpawner.sqx

```diff
@@ -125,12 +125,69 @@ namespace Intrusion.Server
 		    clearBackpackCargoGlobal _vehicleObj;
 		    clearItemCargoGlobal _vehicleObj;
 		    
-		    {
-            	_vehicleObj addBackpackCargoGlobal [_x, 1];
-		    } foreach _vehicleClassInfo.CargoWeapons as String;
-		    
         	_intrusionVehicle = [_vehicleObj, _side, _professionType, _group, _cost] new IntrusionVehicle;
         	_vehicleObj setVariable ["IntrusionVehicle", _intrusionVehicle, true];
+        	
+	    	{
+	    		private ["_attachObj" as Object, "_attachObjInfo" as VehicleAttachmentInfo];
+	    		_attachObjInfo = _x;
+	    		
+	    		_attachObj = _attachObjInfo.ClassName createVehicle (getPos _vehicleObj);
+				_attachObj attachTo [_vehicleObj, _attachObjInfo.AttachOffset];
+				clearWeaponCargoGlobal _attachObj;
+				clearItemCargoGlobal _attachObj;
+				clearMagazineCargoGlobal _attachObj;
+				
+				// Fill the weaponbox again
+				{
+	            	if (_x.ClassName isKindOf ["Bag_Base", configFile>>"CfgVehicles"]) then {
```

Comment:
Better to use _attachObjInfo than _x, since you have it...

Reply:
_attachObjInfo is config information about what object to attach, and the _x on this row is a VehicleCargoInfo item.